### PR TITLE
feat(replay): Show Seen-by list of users on Replay Details

### PR DESCRIPTION
**Pull request by ryan953**
SCR-20240411-nzaf

Avatar list is kinda weird, it sticks out to the left:
SCR-20240411-nzka

Relates to https://github.com/getsentry/team-replay/issues/19
Relates to https://github.com/getsentry/sentry/issues/64924

Depends on https://github.com/getsentry/sentry/pull/68990

### CODE REVIEW

**Review by michellewzhang — APPROVED** [review]
should we check for `isError`or `isLoading` in the `useApiQuery` call or is that not needed here?

**Comment by michellewzhang** [comment]
> I didn't deal with isError because I didn't really know what to tell the user :(. We could offer a retry button, but it's a small part of the page so idk if that's just distracting or calling too much attention to it. Open to ideas though.

that's a good point. placeholder makes sense as a subtle default here, which i think you've already done 👍

File: static/app/utils/replays/hooks/useReplayViewedByData.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import {
+  type ApiQueryKey,
+  useApiQuery,
+  type UseApiQueryOptions,
+} from 'sentry/utils/queryClient';
+
+interface Props {
+  orgSlug: string;
+  projectSlug: undefined | string;
+  replayId: undefined | string;
```
Comment (aliu3ntry) [inline static/app/utils/replays/hooks/useReplayViewedByData.tsx:11]: Why is `undefined` allowed for replayId and projectSlug? Why not for orgSlug?

Comment (michellewzhang) [reply static/app/utils/replays/hooks/useReplayViewedByData.tsx:11]: @aliu3ntry it comes from the `replayMetaData.tsx` file that calls this hook. `replayRecord` might not exist, but that's where we get the `projectSlug` and `replayId` from

`projectSlug: replayRecord?.project_id,`

Comment (aliu3ntry) [reply static/app/utils/replays/hooks/useReplayViewedByData.tsx:11]: Got it, what happens if undefined is passed to the apiQuery? Will we use a network call calling the nonexistent endpoint `"projects/undefined/...."`?

**Comment by billyvg** [comment]
> > should we check for `isError`or `isLoading` in the `useApiQuery` call or is that not needed here?
> 
> I didn't deal with isError because I didn't really know what to tell the user :(. We could offer a retry button, but it's a small part of the page so idk if that's just distracting or calling too much attention to it. Open to ideas though.
> 
> As for loading, it'll just render once we have data. So we don't need to check the other field.

We have a "mini" error boundary component I think, but I agree that this is a small part of the page and not a deal breaker if it's broken (as long as its logged to Sentry)

File: static/app/utils/replays/hooks/useReplayViewedByData.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/utils/replays/hooks/useReplayViewedByData.tsx:7]: fixed the `projects/undefined/....` problem. good catch @andrewshie-sentry!

**Comment by ryan953** [comment]
> We have a "mini" error boundary component I think, but I agree that this is a small part of the page and not a deal breaker if it's broken (as long as its logged to Sentry)

Added logging. The mini variant is still a bit big. But we can iterate on it too depending on how often it happens.

File: static/app/utils/replays/hooks/useReplayViewedByData.tsx (hunk 1 above)
Comment (aliu3ntry) [reply static/app/utils/replays/hooks/useReplayViewedByData.tsx:7]: thanks! wrong andrew :(

**Comment by ryan953** [comment]
> should we check for `isError`or `isLoading` in the `useApiQuery` call or is that not needed here?

@michellewzhang you were right about this. I think the tests in CI are trying to assert that the placeholder goes away, but with the setup i had before they could be there forever. Now it's checking for isLoaded, and if it's loaded we'll remove the placeholder for sure, and render 0 or more avatars.

**Pull request merged**